c2rust-ast-exporter/src/AstExporter.cpp

```diff
@@ -189,6 +189,24 @@ class TypeEncoder final : public TypeVisitor<TypeEncoder> {
           astEncoder(ast) {}
 
     void VisitQualType(const QualType &QT) {
+        if(isa<AtomicType>(QT)) {
```

Comment:
This check needs to be moved inside the `if (!QT.isNull()) {` conditional below; otherwise we get an internal Clang abort doing `isa<AtomicType>(QT)` when QT has a null inner pointer.